### PR TITLE
Update `LearningRateMonitor` docs and tests for `log_weight_decay`

### LINKED ISSUE #19798: Missing documentation for the `log_weight_decay` argument in `lightning.pytorch.callbacks.LearningRateMonitor`
### 📚 Documentation

The [LearningRateMonitor class documentation](https://lightning.ai/docs/pytorch/latest/api/lightning.pytorch.callbacks.LearningRateMonitor.html) currently has no explanation for the `log_weight_decay` argument.

cc @borda

### PR DESCRIPTION
**Pull request by Peiffap**
## What does this PR do?

Fixes #19798.

Adds a description for the `log_weight_decay` argument for `LearningRateMonitor`, fixes another miscellaneous typo in the docs for that class, and adds an additional `assert` statement to the tests for `LearningRateMonitor` to ensure no weight decay values are tracked if `log_weight_decay=False` (by analogy with the tests for (`log_momentum=False`).

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
📚 Documentation preview 📚: https://pytorch-lightning--19805.org.readthedocs.build/en/19805/

### CODE REVIEW

**Comment by Peiffap** [comment]
@Borda Apologies for having to ping you on this; could you share the reason this PR is being held back?

**Review by Borda — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]
